src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -219,7 +219,11 @@ public Map<String, AccountUsageCalculation> collect(Collection<String> products,
     if (log.isDebugEnabled()) {
       calcsByAccount.values().forEach(calc -> log.debug("Account Usage: {}", calc));
     }
-
+    // SWATCH-121 This logic should be implemented much cleaner
+    accountServiceInventory.getServiceInstances().values().stream()
+        .map(Host::getOrgId)
+        .findFirst()
```

Comment:
Apparently, `findFirst()` does not allow nulls [per the javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#findFirst--). Found via https://stackoverflow.com/a/32476601

I played around a bit in `jshell` which made it easy to reproduce:

```
Stream.of(1).map(a -> null).findFirst().ifPresent((a) -> {})
```

fails with a similar stacktrace. You can easily work around by filtering null values out:

```
Stream.of(1).map(a -> null).filter(Objects::nonNull).findFirst().ifPresent((a) -> {})
```

So I would suggest:
```suggestion
        .filter(Objects::nonNull)
        .findFirst()
```
(will need to adjust import statements afterwards).